src/number/n_transExt.jl

```diff
@@ -369,15 +369,16 @@ end
 ###############################################################################
 
 @doc Markdown.doc"""
-    FunctionField(F::Singular.Field, S::Array{String, 1})
+    FunctionField(F::Singular.Field, S::Vector{String})
 
 Returns a tuple $K, a$ consisting of a function field $K$ over the field $F$
 with transcendence basis stored in the array $S$.
 """
-function FunctionField(F::Singular.Field, S::Array{String, 1}; cached::Bool=true)
+function FunctionField(F::Singular.Field, S::Vector{String}; cached::Bool=true)
    typeof(F) == N_GField && error("Finite field extensions of Fp not supported.")
-   l = length(S)
-   isempty(l) && throw(DomainError())
+   isempty(S) && throw(ArgumentError("array must be non-empty"))
+   any(isempty, S) && throw(ArgumentError("strings in array must be non-empty"))
+   allunique(S) && throw(ArgumentError("strings in array must be pairwise different"))
```

Comment:
I think you mean !allunique

Reply:
And then the tests should all pass

Reply:
Yes thank you I also just noticed because of failing tests!

Reply:
Now all tests passed. I will merge.